core/load_balancer_rule_test.go

```diff
@@ -0,0 +1,446 @@
+package core
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/krystal/go-katapult"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestClient_LoadBalancerRules(t *testing.T) {
+	c := New(&fakeRequestMaker{})
+
+	assert.IsType(t, &LoadBalancerRulesClient{}, c.LoadBalancerRules)
+}
+
+func TestLoadBalancerRule_JSONMarshalling(t *testing.T) {
+	tests := []struct {
+		name string
+		obj  *LoadBalancerRule
+	}{
+		{
+			name: "empty",
+			obj:  &LoadBalancerRule{},
+		},
+		{
+			name: "full",
+			obj: &LoadBalancerRule{
+				ID:              "arbitrary string",
+				Algorithm:       StickyRuleAlgorithm,
+				DestinationPort: 1024,
+				ListenPort:      1337,
+				Protocol:        HTTPProtocol,
+				ProxyProtocol:   true,
+				Certificates: []Certificate{
+					{
+						ID: "another abitrary string",
+					},
+				},
+				BackendSSL:     true,
+				PassthroughSSL: true,
+				CheckEnabled:   true,
+				CheckFall:      3,
+				CheckInterval:  50,
+				CheckPath:      "/healthz",
+				CheckProtocol:  HTTPProtocol,
+				CheckRise:      12,
+				CheckTimeout:   3,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			testJSONMarshaling(t, tt.obj)
+		})
+	}
+}
+
+func TestLoadBalancerRuleArguments_JSONMarshalling(t *testing.T) {
+	tests := []struct {
+		name string
+		obj  *LoadBalancerRuleArguments
+	}{
+		{
+			name: "empty",
+			obj:  &LoadBalancerRuleArguments{},
+		},
+		{
+			name: "full",
+			obj: &LoadBalancerRuleArguments{
+				Algorithm:       StickyRuleAlgorithm,
+				DestinationPort: 1024,
+				ListenPort:      1337,
+				Protocol:        HTTPProtocol,
+				ProxyProtocol:   true,
+				Certificates: []Certificate{
+					{
+						ID: "another abitrary string",
+					},
+				},
+				CheckEnabled:  true,
+				CheckFall:     3,
+				CheckInterval: 50,
+				CheckPath:     "/healthz",
+				CheckProtocol: HTTPProtocol,
+				CheckRise:     12,
+				CheckTimeout:  3,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			testJSONMarshaling(t, tt.obj)
+		})
+	}
+}
+
+func TestLoadBalancerRulesClient_Delete(t *testing.T) {
```

Comment:
nitpick: I like placing the tests in the same order as the functions they're testing, just so the placement/position between the implementation and test files correlate 😁